Source/VMware.vSphereDSC/DSCResources/VMHostNetwork/VMHostVss/VMHostVssSecurity.ps1

```diff
@@ -0,0 +1,169 @@
+
+            return ($null -eq $vss -or $this.Equals($vss))
+        }
+    }
+
+    [VMHostVssSecurity] Get() {
+        Write-Verbose -Message "$(Get-Date) $($s = Get-PSCallStack; "Entering {0}" -f $s[0].FunctionName)"
+
+        $result = [VMHostVssSecurity]::new()
+        $result.Server = $this.Server
+
+        $this.ConnectVIServer()
+        $vmHost = $this.GetVMHost()
+        $this.GetNetworkSystem($vmHost)
+
+        $result.Name = $vmHost.Name
+        $this.PopulateResult($vmHost, $result)
+
+        if ('' -ne $result.VssName) {
+            $result.Ensure = 'Present'
+        }
+        else {
+            $result.Ensure = 'Absent'
+        }
+
+        return $result
+    }
+
+    <#
+    .DESCRIPTION
+
+    Returns a boolean value indicating if the VMHostVssSecurity should to be updated.
+    #>
+    [bool] Equals($vss) {
+        Write-Verbose -Message "$(Get-Date) $($s = Get-PSCallStack; "Entering {0}" -f $s[0].FunctionName)"
+
+        $vssSecurityTest = @()
+        $vssSecurityTest += ($vss.Spec.Policy.Security.AllowPromiscuous -eq $this.AllowPromiscuous)
+        $vssSecurityTest += ($vss.Spec.Policy.Security.ForgedTransmits -eq $this.ForgedTransmits)
+        $vssSecurityTest += ($vss.Spec.Policy.Security.MacChanges -eq $this.MacChanges)
+
+        return ($vssSecurityTest -notcontains $false)
+    }
+
+    <#
+    .DESCRIPTION
+
+    Updates the configuration of the virtual switch.
+    #>
+    [void] UpdateVssSecurity($vmHost) {
+        Write-Verbose -Message "$(Get-Date) $($s = Get-PSCallStack; "Entering {0}" -f $s[0].FunctionName)"
+
+        $vssSecurityArgs = @{
+            Name = $this.VssName
+            AllowPromiscuous = $this.AllowPromiscuous
+            ForgedTransmits = $this.ForgedTransmits
+            MacChanges = $this.MacChanges
+        }
+        $vss = $this.GetVss($vmHost)
+
+        if ($this.Ensure -eq 'Present') {
+            if ($this.Equals($vss)) {
+                return
+            }
+            $vssSecurityArgs.Add('Operation', 'edit')
+        }
+        else {
+            $vssSecurityArgs.AllowPromiscuous = $false
+            $vssSecurityArgs.ForgedTransmits = $true
+            $vssSecurityArgs.MacChanges = $true
+            $vssSecurityArgs.Add('Operation', 'edit')
+        }
+
+        try {
+            Update-Network -NetworkSystem $this.vmHostNetworkSystem -VssSecurityConfig $vssSecurityArgs -ErrorAction Stop
+        }
+        catch {
+            Write-Error "The Virtual Switch Security Config could not be updated: $($_.Exception.Message)"
+        }
+    }
+
+    <#
+    .DESCRIPTION
+
+    Populates the result returned from the Get() method with the values of the Security settings of the Virtual Switch.
+    #>
+    [void] PopulateResult($vmHost, $vmHostVSSSecurity) {
+        Write-Verbose -Message "$(Get-Date) $($s = Get-PSCallStack; "Entering {0}" -f $s[0].FunctionName)"
+
+        $currentVss = $this.GetVss($vmHost)
+
+        if ($null -ne $currentVss) {
```

Comment:
Add the else clause here like in the VMHostVss resource.

Reply:
This is slightly different imho.

We can only set the VssName property (inherited from VMHostVssBase) to the requested VSS.
All the other properties make no sense since the VSS does not exist.

Perhaps we should move the VssKey property from VMHostVss to VMHostVssBaseDSC?
Same as VssName.

Reply:
Well yes in the else clause you could only add the VssName property to be equal to $this.VssName.
For now do not move the VssKey property.

So only add the else clause with the VssName in it.